drivers/snmp-ups.c

```diff
@@ -3504,19 +3506,26 @@ bool_t su_ups_get(snmp_info_t *su_info_p)
 
 	if (!strcasecmp(su_info_p->info_type, "ups.status")) {
 /* FIXME: daisychain status support! */
-		upsdebugx(2, "%s: requesting nut_snmp_get_int() for "
-			"ups.status, with%s daisy template originally",
-			__func__, (format_char!=NULL ? "" : "out"));
-		status = nut_snmp_get_int(su_info_p->OID, &value);
-		if (status == TRUE)
+		if(!(strcasecmp(su_info_p->OID, ".1.3.6.1.4.1.318.1.1.1.11.1.1.0")))
 		{
-			su_status_set(su_info_p, value);
-			upsdebugx(2, "=> value: %ld", value);
+			status = nut_snmp_get_str(su_info_p->OID, buf, sizeof(buf), su_info_p->oid2info);
+			char pwr_status;
```

Comment:
This looks error-prone: `strtol()` deals with a (nul-terminated) string, but your copy of the first character from `buf` is not necessarily followed in RAM by a zero byte. Any random garbage can pop up here...

Trying to make sense of that MIB/OID with http://oidref.com/1.3.6.1.4.1.318.1.1.1.11.1.1.0 I see that it is a child of http://oidref.com/1.3.6.1.4.1.318.1.1.1.11.1.1 but otherwise got no clue of its own value. That parent however is a string representing 64 bit-flags (or `UNKNOWN` which you don't check for), and the second byte (your `buf[1]`?) would be the "On Battery" flag. There is a separate flag for "On Line" as well as bypass, self-test, and a united "Low Battery / On Battery" flag, so it may be prudent to check several values for this reading.

Also wondering if all this complex logic should be here or can be better served by "lookup functions" in the `*-mib.c` file (the later, generally `NULL`, fields of lookup tables like `APCC_G5500_pwr_info[]`)?.. See e.g. `drivers/eaton-pdu-marlin-mib.c` for some practical examples.

Reply:
I'll try to learn about `strtol()` and its functioning, thank you.

As far as I know, those lookup functions only work when (to put it simply) 
the macro `WITH_SNMP_LKP_FUN' is set to 1

drivers/snmp-ups.h  137:142
```
# if (defined WITH_DMFMIB) && (WITH_DMFMIB != 0)
#  define WITH_SNMP_LKP_FUN 0
# else
#  define WITH_SNMP_LKP_FUN 1
# endif
#endif
```

Does it mean that in best scenario I will still have to use a self-built version of NUT for this particular UPS model?
Or I might be missing something 

Yes, I acknowledge the fact that `ups.status` is more than `OL` or `OB`, things would get easier with lookup functions, I understand

Reply:
And umm I have put child OID (not parent) there because that's what actually works when using something like `snmpget`
The parent OID does not work

`snmpget -v1 -c public 'redacted_ip' 1.3.6.1.4.1.318.1.1.1.11.1.1`
```
Error in packet
Reason: (noSuchName) There is no such variable name in this MIB.
Failed object: iso.3.6.1.4.1.318.1.1.1.11.1.1
```